bluesky/bundlers.py

```diff
@@ -620,18 +600,84 @@ async def complete(self, msg):
+        """
+
+        # If there's a message_stream_name, check the describe_collect returned acceptable output for it
+        if message_stream_name:
+            # The collect contained a name and describe_collect returned a Dict[str, Dict[str, DataKey]],
+            # this is only acceptable if the only key in the parent dict is message_stream_name
+            if len(describe_collect_dict) == 1:
+                assert message_stream_name in describe_collect_dict
+                return describe_collect_dict.items()
+
+            # We ensure that describle_collect returned a Dict[str, DataKey], corresponding
+            # to the message_stream_name, or that the describe_collect includes the message_stream_name
+            # DataKey dict.
+            if len(describe_collect_dict):
+                # In future this should be changed to validate against a new event_model schema Dict[str, DataKey]
+                # event_model.schemas.data_key so this code will change with the schema, the below checks to see
+                # if the Required keys are the correct form.
+                should_be_DataKey = list(describe_collect_dict.values())[0]
+
+                if "source" in should_be_DataKey:
+                    assert isinstance(should_be_DataKey["source"], str), \
+                        "Expected describe_collect output to be of type Dict[str, DataKey] for the descriptor of "\
+                        f"the passed in {message_stream_name}"
+                    return [(message_stream_name, describe_collect_dict)]
+                else:
+                    assert message_stream_name in describe_collect_dict, "A name was supplied, "\
+                        "but this name was not present in the describe_collect keys"
+
+            else:
+                raise RuntimeError(
+                    f"Invalid describe_collect return: {describe_collect_dict} when collect "
+                    f"was called on {message_stream_name}"
+                )
+        # If there was no name kwarg to the collect Msg, raise an error if the dict isn't
+        # Dict[str, Dict[str, DataKey]]
+        else:
+            should_be_a_dict_to_data_key = describe_collect_dict[next(iter(describe_collect_dict))]
+            assert isinstance(should_be_a_dict_to_data_key, dict),\
+                "describe_collect() returned a Dict[str, SomeTypeNotADict], this would only be acceptable "\
+                "if a name argument was included in the collect Msg"
+
+            if should_be_a_dict_to_data_key:
+                should_be_a_data_key = should_be_a_dict_to_data_key[next(iter(should_be_a_dict_to_data_key))]
+                assert "source" in should_be_a_data_key and isinstance(should_be_a_data_key["source"], str),\
+                    "describe_collect() returned a Dict[str, Dict[str, SomeTypeNotADataKey]]"
+
+            return describe_collect_dict.items()
```

Comment:
This logic didn't work for my use case of `Dict[str, DataKey]` with a single key and the message stream name given, it fired the first assert statement on line 622. This logic worked for me:

```suggestion
    def _maybe_format_datakeys_with_stream_name(
        self,
        describe_collect_dict: Union[Dict[str, DataKey], Dict[str, Dict[str, DataKey]]],
        message_stream_name: Optional[str] = None
    ) -> List[Tuple[str, Dict[str, DataKey]]]:
        """
        Check if the dictionary returned by describe collect is a dict
            `{str: DataKey}` or a `{str: {str: DataKey}}`.

        If a `message_stream_name` is passed then return a singleton list of the form of
            `{message_stream_name: describe_collect_dict}.items()`.
        If the `message_stream_name` is None then return the `describe_collect_dict.items()`.
        """
        def has_str_source(d: dict):
            return isinstance(d, dict) and isinstance(d.get("source", None), str)

        if describe_collect_dict:
            first_value = list(describe_collect_dict.values())[0]
            if has_str_source(first_value):
                # We have Dict[str, DataKey], so return just this
                # If stream name not given then default to "primary"
                return [(message_stream_name or "primary", describe_collect_dict)]
            elif all(has_str_source(v) for v in first_value.values()):
                # We have Dict[str, Dict[str, DataKey]] so return its items
                if message_stream_name and list(describe_collect_dict) != [message_stream_name]:
                    # The collect contained a name and describe_collect returned a Dict[str, Dict[str, DataKey]],
                    # this is only acceptable if the only key in the parent dict is message_stream_name
                    raise RuntimeError(
                        f"Expected a single stream {message_stream_name!r}, got {describe_collect_dict}"
                    )
                return list(describe_collect_dict.items())
            else:
                raise RuntimeError(
                    f"Invalid describe_collect return: {describe_collect_dict} when collect "
                    f"was called on {message_stream_name}"
                )
        else:
            # Empty dict, could be either but we don't care
            return []
```